interop/xds/server/server.go

```diff
@@ -78,8 +87,48 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID}
+
+	for _, val := range getRPCBehaviorMetadata(ctx) {
+		if st := getStatusForRPCBehaviorMetadata(val); st != nil {
+			return response, status.Err(st.Code(), st.Message())
+		}
+	}
+
+	response.Hostname = s.hostname
 	grpc.SetHeader(ctx, metadata.Pairs("hostname", s.hostname))
-	return &testpb.SimpleResponse{ServerId: s.serverID, Hostname: s.hostname}, nil
+	return response, status.Err(codes.OK, "")
+}
+
+func getRPCBehaviorMetadata(ctx context.Context) []string {
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		logger.Error("failed to receive metadata")
+		return nil
+	}
+
+	mdRPCBehavior := md.Get(rpcBehaviorMetadataKey)
+
+	rpcBehaviorMetadata := make([]string, 0)
```

Comment:
What is the point of allocating a slice with a capacity of `0`. Instead you would do `var rpcBehaviorMetadata []string` or `rpcBehaviorMetadata := []string{}`. 

And since `append` works on `nil` slices, the first option is totally viable.

Reply:
I forgot to refactor the initialisation when I changed it from map to slice, will correct it now